lib/schema.js

```diff
@@ -540,6 +540,12 @@ Schema.prototype.add = function add(obj, prefix) {
         this.nested[prefix.substring(0, prefix.length - 1)] = true;
       }
       this.path(prefix + key, val);
+      if (utils.isPOJO(val[0]) && utils.isPOJO(val[0].discriminators)) {
```

Comment:
```suggestion
      if (utils.isPOJO(val[0] && val[0].discriminators)) {
```

Reply:
@vkarpov15 
Would it be more intention revealing to use
`if ((val[0] instanceof Schema) === false && val[0].discriminators) {`?

Reply:
That's fair, we can do `&& !(val[0] instanceof Schema)` instead